### ISSUE TITLE
Draw figure's icon using an icon font instead of code

### ISSUE THREAD

**Issue by jbsarrodie** [archimatetool/archi#527]
See #522

As of now, figure's icon is drawn using custom code. Another option would be to use an icon font. This would have several advantages:
- There are several free/open icon fonts available, making it easy to extend the set of icons to allow people to use non default notation. This opens the door for language customization and stereotyping
- There are good and free tools online (like [icomoon)[https://icomoon.io/app/#/select]) that provide an easy way to import SVG and convert it to a font. This is an easy path to add new icons when needed.
- There is now some fonts that use two colors or two shades. This might be helpful (if supported) to have even beautiful icons ;-)

**Comment by herve91** [comment]
Hi JB,

This would simplify a lot my specialization plugin if Archi could natively replace figure's icons by custom ones :-)

Best regards
Hervé

**Comment by jbsarrodie** [comment]
>This would simplify a lot my specialization plugin if Archi could natively replace figure's icons by custom ones :-)

Yes. In fact my long term goal is to add native support for language customization in Archi. So this is a first step.

**Comment by herve91** [comment]
Did you have a chance to have a look to my alpha version of my specialization plugin v2 ?

Unluckily, I lack time for this plugin but the direction I took is to define specializations at the model level, a specialization being composed of a an image, graphical configuration (color, line size, ...), and a set of properties.

Then Archimate concept by Archimate concept, add the ability to assign a specialization which automatically sets the image, the graphical configuration and the properties.

If this is close to what you've got in mind and if Archi implements this natively, then my specialization plugins will not be required anymore :-)

**Comment by jbsarrodie** [comment]
Yes, that's what I have in mind.

**Comment by herve91** [comment]
Great :-)

**Comment by Phillipus** [comment]
The thing with setting icons and images is making sure they are bundled with the *.archimate file.

**Comment by jbsarrodie** [comment]
> The thing with setting icons and images is making sure they are bundled with the *.archimate file.

Not fully true. If we provide an icon font (containing standard ArchiMate icons and a good set of other icons) with Archi, then we don't have to bundle icons with the .archimate file. Of course, if we want to support real images, then they have to be bundles. But in this case, we can already do it (we have images in canvas and Hervé's plugin uses this trick to store custom images).

**Comment by jbsarrodie** [comment]
> Yes, that's what I have in mind.

And BTW, don't expect it to be implemented anytime soon. There is a lot of discussions and design decisions involved prior any code.

**Comment by dereksz** [comment]
FYI: Both Font Awsome and Nerd Fonts (https://www.nerdfonts.com/) may be useful references / starting points for custom fonts.
